frontend/src/component/Account/PersonalData.jsx

```diff
@@ -78,7 +83,7 @@ export class PersonalData extends React.Component {
               placeholder={props.userPublicName}
               value={state.newPublicName}
               onChange={this.handleChangePublicName}
-              disabled={!shouldAllowEditionFromAuthType}
+              disabled={this.isFieldWritable('public_name')}
```

Comment:
disable=isFieldWritable(xxx) ?

I suggest a `isReadOnly()` function which is more straightforward to understand `disabled={this.isReadOnly('public_name')}`

Reply:
Actually, I do not understand: if the field is writable, then `isFieldWritable(xxx)` returns True, then disabled=true which is wrong

Reply:
Indeed I mixed-up the naming and function... I will make it more clear with something like `isReadOnly`

Reply:
Thinking about it. It really checks if it is editable, not just if it is read-only (an admin can write, despite the field being read-only).

I will adjust the naming and return value accordingly.

Reply:
I changed it to `isDisabled`, this makes it more clear about what it returns and checks